src/HW/client/src/components/ListControls/ListControls.jsx

```diff
@@ -28,9 +32,9 @@ const ListControls = ({ renderNewItems }) => {
   };
   const handlePressClick = async () => {
     try {
-      setLoading(true);
-      await renderNewItems(inputValue);
-      setLoading(false);
+      showLoaderAction();
+      await addItemsAction(inputValue);
+      hideLoaderAction();
```

Comment:
you can inserts all 3 to the addItemsAction
it will first dispatch an action of starting the fetch, then after success/failure will dispatch the relevant action that will change the state - both the items and the loading status